commit 14571f123ad56909986f795c26ae48e42b0ceb26
Author: Raphael Kubo da Costa <rakuco@FreeBSD.org>
Date:   Sat Aug 31 16:12:49 2013 +0300

    Use g_mkdir() instead of different versions of mkdir().
    
    The current code was erroneously relying on the _POSIX_C_SOURCE macro to
    decide whether to use mkdir() with the signature present in POSIX-compliant
    systems or with the Windows signature.
    
    It turns out POSIX.1 expects the applications to define _POSIX_C_SOURCE to
    certain values. Only glibc resorts to defining it to a value by default,
    opposing what the standard says. It would then break systems that use
    another libc, such as the BSDs, OS X and maybe even other Linux libc
    implementations.
    
    Work around the issue by just relying on glib's g_mkdir(), which has the
    proper means to decide which mkdir() function to call. We still need to
    check for whether we are on Windows, though, since some of the mode
    constants are not defined on it.
    
    REVIEW:		112402

diff --git a/src/oxygenqtsettings.cpp b/src/oxygenqtsettings.cpp
index 9027d30..10ebdf8 100644
--- src/oxygenqtsettings.cpp
+++ src/oxygenqtsettings.cpp
@@ -28,6 +28,8 @@
 #include "oxygentimeline.h"
 #include "config.h"
 
+#include <glib.h>
+#include <glib/gstdio.h>
 #include <gtk/gtk.h>
 
 #include <algorithm>
@@ -38,7 +40,6 @@
 #include <fstream>
 #include <iostream>
 #include <sstream>
-#include <unistd.h>
 
 namespace Oxygen
 {
@@ -330,11 +331,12 @@ namespace Oxygen
         struct stat st;
         if( stat( _userConfigDir.c_str(), &st ) != 0 )
         {
-
-            #if _POSIX_C_SOURCE
-            mkdir( _userConfigDir.c_str(), S_IRWXU|S_IRWXG|S_IRWXO );
+            #ifdef G_OS_WIN32
+            // S_IRWXG and S_IRWXO are undefined on Windows, and g_mkdir()
+            // ignores its second parameter on Windows anyway.
+            g_mkdir( _userConfigDir.c_str(), 0 );
             #else
-            mkdir( _userConfigDir.c_str() );
+            g_mkdir( _userConfigDir.c_str(), S_IRWXU|S_IRWXG|S_IRWXO );
             #endif
         }
 
